components/VM_Arm/src/main.c

```diff
@@ -719,15 +721,42 @@ static int generate_fdt(vm_t *vm, void *fdt_ori, void *gen_fdt, int buf_size, si
                         int num_paths)
 {
     int err = 0;
+
+    int num_keep_devices;
+    char **keep_devices;
+    if (camkes_dtb_get_plat_keep_devices) {
+        keep_devices = camkes_dtb_get_plat_keep_devices(&num_keep_devices);
+    }
+
+    int num_keep_devices_and_subtree;
```

Comment:
num_keep_devices and num_keep_devices_and_subtree should be initialized to 0 in case camkes_dtb_get_plat_keep_devices is null and they don't get initialized.

Reply:
Good catch. Done.